nativeunwind/elfunwindinfo/elfgopclntab.go

```diff
@@ -474,10 +470,37 @@ func (ee *elfExtractor) parseGoPclntab() error {
 	// would fill up our precious kernel delta maps fast, the strategy is to
 	// create deltastack maps for non-Go source files only, and otherwise
 	// cover the vast majority with "use frame pointer" stack delta.
-	sourceStrategy := make(map[int]int)
+	sourceStrategy := make(map[int]strategy)
 
 	// Get target machine architecture for the ELF file
 	arch := ef.Machine
+	defaultStrategy := strategyFramePointer
+	var parsePclntab func(*sdtypes.StackDeltaArray, *pclntabFunc, uintptr, []byte,
+		strategy, uint64, uint8) error
+
+	switch arch {
+	case elf.EM_X86_64:
+		parsePclntab = parseX86pclntabFunc
+	case elf.EM_AARCH64:
+		parsePclntab = parseArm64pclntabFunc
+		// Go 1.20 and earlier did not maintain frame pointers properly on arm64.
+		// This was fixed for Go 1.21 and later in:
+		// https://github.com/golang/go/commit/a41a29ad19c25c3475a65b7265fcad870d954c2a
+		switch hdr.magic {
+		case magicGo1_2, magicGo1_16, magicGo1_18:
+			// Magic indicates old Go with broken arm64 frame pointers
+			defaultStrategy = strategyDeltasWithFrame
+		case magicGo1_20:
+			// Ambiguous regarding if frame pointer is kept correctly.
+			// Take the slow path of resolving Go version.
+			goVer, err := ee.file.GoVersion()
+			if err != nil || version.Compare(goVer, "go1.21rc1") < 0 {
```

Comment:
Nit: Release candidates are for temporary testing. Maybe we can just use the first release with this change in Go:
```suggestion
			if err != nil || version.Compare(goVer, "go1.21.0") < 0 {
```

Reply:
I have a strong preference to keep the actual first rc tag here. The Go libraries support comparing versions with them, and this makes the code work with the prerelease tags. This also serves as note to remind what is the first tag when the Golang change was made.